enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,206 @@
+
+### Workflow Description
+
+- To enroll a `MachineSet` for boot image updates, the cluster admin should add an annotation `io.openshift.mco-managed=true` to the `MachineSet`.
+- To un-enroll(and effectively rollback) the `MachineSet` from boot image updates, the cluster admin should remove the `io.openshift.mco-managed=true` annotation from the `MachineSet`.
+
+#### Variation and form factor considerations [optional]
+
+Any form factor using the MCO and `MachineSets` will be impacted by this proposal. So case by case:
+- Standalone OpenShift: Yes, this is the main target form factor.
+- microshift: No, as it does [not](https://github.com/openshift/microshift/blob/main/docs/contributor/enabled_apis.md) use `MachineSets`.
+- Hypershift: No, Hypershift does not have this issue.
+
+### API Extensions
+
+We may have to make some changes to MCO CRDs for the opt-in feature.
+
+### Implementation Details/Notes/Constraints [optional]
+
+![Sub Controller Flow](manage_boot_images_flow.jpg)
+
+![MachineSet Reconciliation Flow](manage_boot_images_reconcile_loop.jpg)
+
+The implementation has a GCP specific POC here:
+- https://github.com/openshift/machine-config-operator/pull/3980
+
+Possible constraints:
+- Ignition spec 2 to spec 3 is not deterministic. Some translations are unsupported and as a result not all stub secrets can be managed. In these cases, failure will be reported, and it will cause a cluster degrade.
+- See Open questions below for some more possible constraints.
+
+### Risks and Mitigations
+
+The biggest risk in this enhancement would be delivering a bad boot image. To mitigate this, we have outlined a rollback option.
+
+How will security be reviewed and by whom? TBD
+This is a solution aimed at reducing usage of outdated artifacts and should not introduce any security concerns that do not currently exist. 
+
+How will UX be reviewed and by whom? TBD 
+The UX element involved include the user opt-in and opt-out, which is currently up for debate. 
+
+### Drawbacks
+
+TBD, based on the open questions below.
+
+## Design Details
+
+### Open Questions
+
+- Should we have a like a global switch that opt-in all `MachineSets` for this mechanism?
+- Somewhat related to above, would we also want to allow opting out without rolling back? This is for a situation for the customer would not want to update the boot images any longer, but would like to keep the current image instead of the "factory" after rolling back. Not sure if anyone would use this, but though it was worth considering.
+- This proposal relies on the golden configmap having a target value for every platform/arch combination that we use today. I've [noticed](https://issues.redhat.com/browse/MCO-793) some cases like vsphere don't have a reference as it stands today. Why is that? Are there scenarios not requiring boot image updates?
```

Comment:
vSphere uses a template VM created by uploading the OVA(?) image to the vCenter. You would need to upload a new image to each vCenter, create it as a template, and then update the template references.
This becomes even harder when you factor the work of @rvanderp3 to add support for vSphere across multiple zones, where the template then comes from the Infrastructure object

Reply:
I think vsphere will definitely need more design, so I think we should call out that in phases